tests/pyfunc/test_backward_compatibility.py

```diff
@@ -0,0 +1,9 @@
+import pytest
+
+import mlflow
+
+
+@pytest.mark.parametrize("version", ["2.7.1", "2.8.1"])
+def test_backward_compatibility(version):
+    model = mlflow.pyfunc.load_model(f"tests/resources/pyfunc_models/{version}")
+    assert model.predict("MLflow is great!") == "MLflow is great!"
```

Comment:
/ai I need your help. Please write that that does the following:

1. Create a temporary directory using tmp_path fixture
2. Save the following code as save.py in that directory
3. Create a docker container with python 3.8
4. Install a specified version of mlflow in the container
5. Run the script.
6. Load the model with `mlflow.pyfunc.load_model`
7. Assert the predict reuslt.

```python
save.py

import mlflow

class MyModel(mlflow.pyfunc.PythonModel):
    def predict(self, context, model_input):
        return model_input

model = MyModel()
mlflow.pyfunc.save_model(
    python_model=model, 
    path="model"
)
```

Reply:
i think this is essentially what @BenWilson2 is doing in https://github.com/mlflow/mlflow/pull/10626/files#diff-0ff0c4b1146adee8062dbda677190fbf8e68095a1bebaa84e561afa29c75e520R1490

Reply:
Not following why this is necessary. It seems sufficient to check in the model files and then try to load them from the current MLflow installation on dev.

Reply:
I agree, let's use locally saved files.